typed-protocols/src/Network/TypedProtocol/Pipelined.hs

```diff
@@ -77,10 +105,46 @@ data PeerReceiver (pk :: PeerKind) (st :: ps) (st' :: ps) m where
             -> (forall st''. Message st st'' -> PeerReceiver pk st'' st' m)
             -> PeerReceiver pk st st' m
 
+effect'
+  :: m (PeerReceiver pk st st' m)
+  ->    PeerReceiver pk st st' m
+effect' = Effect'
+
+await
+  :: (CurrentAgency pk (AgencyInState st) ~ Awaiting)
+  => StateToken st
+  -> (forall st''. Message st st'' -> PeerReceiver pk st'' st' m)
+  -> PeerReceiver pk st st' m
+await = Await
+
 data ReceiveHandler pk ps m where
      ReceiveHandler :: PeerReceiver pk (st :: ps) (st' :: ps) m
                     -> ReceiveHandler pk ps m
 
+-- |
+-- Like `Network.Protocol.Core.connect`, it is also a partial function.
```

Comment:
Bad reference? Are you referring to `Protocol.Core` from `typed-transitions`? That function is not partial. Why should we be OK with including this partial function? Doesn't it undermine the entire point of this development?